src/Authorization/FitOnFhir.Authorization/AuthorizationFunction.cs

```diff
@@ -6,35 +6,63 @@
 using System.Threading;
 using System.Threading.Tasks;
 using EnsureThat;
+using FitOnFhir.Authorization.Handlers;
 using FitOnFhir.Authorization.Services;
+using FitOnFhir.Common.Config;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Azure.WebJobs;
 using Microsoft.Azure.WebJobs.Extensions.Http;
 using Microsoft.Extensions.Logging;
+using ExecutionContext = Microsoft.Azure.WebJobs.ExecutionContext;
 
 namespace FitOnFhir.Authorization
 {
     public class AuthorizationFunction
     {
         private readonly IRoutingService _routingService;
+        private readonly IFitOnFhirAuthenticationHandler _authenticationHandler;
+        private readonly AuthenticationConfiguration _authenticationConfiguration;
         private readonly ILogger _logger;
 
-        public AuthorizationFunction(IRoutingService routingService, ILogger<AuthorizationFunction> logger)
+        public AuthorizationFunction(
+            IRoutingService routingService,
+            IFitOnFhirAuthenticationHandler authenticationHandler,
+            AuthenticationConfiguration authenticationConfiguration,
+            ILogger<AuthorizationFunction> logger)
         {
-            _routingService = EnsureArg.IsNotNull(routingService);
+            _routingService = EnsureArg.IsNotNull(routingService, nameof(routingService));
+            _authenticationHandler = EnsureArg.IsNotNull(authenticationHandler, nameof(authenticationHandler));
+            _authenticationConfiguration = EnsureArg.IsNotNull(authenticationConfiguration, nameof(authenticationConfiguration));
             _logger = EnsureArg.IsNotNull(logger);
         }
 
         [FunctionName("api")]
         public async Task<IActionResult> Run(
             [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "{p1?}/{p2?}/{p3?}")] HttpRequest req,
-            Microsoft.Azure.WebJobs.ExecutionContext context,
+            ExecutionContext context,
             CancellationToken cancellationToken)
         {
+            bool isAuthenticated = true;
+
             _logger.LogInformation("incoming request from: {0}", req.Host + req.Path);
             using var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, req.HttpContext.RequestAborted);
-            return await _routingService.RouteTo(req, context, cancellationSource.Token);
+
+            if (!_authenticationConfiguration.IsAnonymousLoginEnabled)
```

Comment:
I think we should encapsulate the `if (!_authenticationConfiguration.IsAnonymousLoginEnabled)` logic in the `FitOnFhirAuthenticationHandler`, just in case this needs to me moved to a different class. This also has a few added benefits:

* The `AuthenticationConfiguration` only needs to be injected into the `FitOnFhirAuthenticationHandler`, rather than multiple classes.
* The entire token validation (including no token/anonymous) is executed by calling one method `AuthenticateToken()`
* Everything token validation test related will most likely end up in one class.
* If we need to move where this validation takes place, the code changes are minimal.

I am a huge fan of the Single Responsibility Principles (part of SOLID), and I'm viewing classes through this lens:

***FitOnFhirAuthenticationHandler** is responsible for ensuring the credentials provided in a request meet the credential requirements.*

***AuthorizationFunction** is responsible for delegating requests.*

Viewed through this lens, you can see that some of the responsibilities of FitOnFhirAuthenticationHandler are leaking into the AuthorizationFunction.